dockerfiles/Dockerfile-rosetta

```diff
@@ -1,50 +1,309 @@
+ENV OPAMKEEPBUILDDIR false
+ENV OPAMREUSEBUILDDIR false
+
+ENV PATH "$PATH:/usr/lib/go/bin:$HOME/.cargo/bin"
+
+# git will clone into an empty dir, but this also helps us set the workdir in advance
+RUN git clone \
+  -b "${CODA_BRANCH}" \
+  --depth 1 \
+  --shallow-submodules \
+  --recurse-submodules \
+  https://github.com/CodaProtocol/coda ${HOME}/${CODA_DIR}
+
+WORKDIR $HOME/$CODA_DIR
+
+ENV OPAMYES 1
+# TODO: handle this opam work without cloning the full repository (directly pull src/opam.export)
+# Installs uri.2.2.1 and its dependencies so that build succeeds, then installs the rest of the deps
+RUN opam install \
+  uri.2.2.1 \
+  base.v0.12.0 \
+  stdio.v0.12.0 \
+  ppx_sexp_conv.v0.12.0 \
+  ppxlib.0.8.1 \
+  ocaml-compiler-libs.v0.11.0 \
+  ocaml-migrate-parsetree.1.5.0 \
+  dune.2.5.1 \
+  ocamlfind.1.8.0 \
+  base-bytes.base \
+  ppx_derivers.1.2.1 \
+  re.1.9.0 \
+  result.1.3 \
+  stringext.1.5.0 \
+  && opam switch import src/opam.export \
+  && sudo chmod -R u+rw ~/.opam
+
+RUN eval $(opam config env) \
+  && opam pin add src/external/ocaml-extlib \
+  && opam pin add src/external/ocaml-sodium \
+  && opam pin add src/external/rpc_parallel \
+  && opam pin add src/external/digestif \
+  && opam pin add src/external/async_kernel \
+  && opam pin add src/external/coda_base58 \
+  && opam pin add src/external/graphql_ppx \
+  && opam clean --logs -cs
+
+
+# Our pins
+#RUN eval $(opam config env) \
+  #&& opam clean --logs -csy --unused-repositories
+  #&& rm -rf src/external
+
+#################################################################################################
+# The "builder" Stage
+# - builds coda and any other binaries required to run a node
+# - should not include any data related to joining a specific network, only the node software itself
+#################################################################################################
+FROM opam-deps AS builder
+
+ARG DUNE_PROFILE=testnet_postake_medium_curves
+
+# Set up environment again for running as opam user, to make sure its working properly
+#USER opam
+#ENV HOME /home/opam
+#WORKDIR ${HOME}/${CODA_DIR}
+
+ENV PATH "$PATH:/usr/lib/go/bin:$HOME/.cargo/bin"
+
+# Build libp2p_helper and clear go module caches
+RUN cd src/app/libp2p_helper \
+   && rm -rf result \
+   && mkdir -p result/bin \
+   && cd src \
+   && go mod download \
+   && cd generate_methodidx \
+   && go build -o ../../result/bin/generate_methodidx \
+   && cd ../libp2p_helper \
+   && go build -o ../../result/bin/libp2p_helper \
+   && go clean --cache --modcache --testcache -r
+
+# Make rosetta-crucial components and the generate_keypair tool
+RUN eval $(opam config env) \
+  && dune build --profile=${DUNE_PROFILE} \
+    src/app/generate_keypair/generate_keypair.exe \
+    src/app/cli/src/coda.exe \
+    src/app/archive/archive.exe \
+    src/app/rosetta/rosetta.exe \
+  && mv _build/default/src/app $HOME/app \
+  && rm -rf _build
+
+#RUN eval $(opam config env) && make deb
```

Comment:
If you want to keep this here, could you add a note as to why it's commented out right now